tests/compare-viewer-reports

```diff
@@ -0,0 +1,218 @@
+
+def stringify(cmd):
+    """Render a command as a string."""
+
+    return ' '.join([str(word) for word in cmd])
+
+def truncate(string, verbose=False, length=800):
+    """Truncate a long string like a json blob."""
+
+    if not verbose and len(string) > length:
+        return string[:800].strip() + ' ...[truncated]...'
+    return string.strip()
+
+def compare_diff(path1, path2, verbose):
+    """Use diff to compare two files or two directories."""
+
+    cmd = ['diff', '-r', path1, path2]
+    result = run(cmd)
+    if result.returncode:
+        print(stringify(cmd))
+        print(truncate(result.stdout, verbose))
+    return result.returncode == 0
+
+def compare_result(path1, path2):
+    """Use compare-result to compare two copies of viewer-result.json."""
+
+    cmd = [SCRIPT_DIR/'compare-result', path1, path2]
+    try:
+        return run(cmd).returncode == 0
+    except UserWarning as err:
+        print(stringify(cmd))
+        print(err)
+        return False
+
+def compare_source(path1, path2):
+    """Use compare-source to compare two copies of viewer-source.json."""
+
+    cmd = [SCRIPT_DIR/'compare-source', path1, path2]
+    try:
+        return run(cmd).returncode == 0
+    except UserWarning as err:
+        print(stringify(cmd))
+        print(err)
+        return False
+
+def compare_html(proof, reports1, reports2,
+                 report='report', html='html', verbose=False):
+    # pylint: disable=too-many-arguments
+    """Compare html reports produced by cbmc-viewer."""
+
+    html_path = proof/report/html
+    if not (reports1/html_path).is_dir():
+        raise FileNotFoundError(reports1/html_path)
+    if not (reports2/html_path).is_dir():
+        raise FileNotFoundError(reports2/html_path)
+
+    logging.info("Comparing %s", html_path)
+    return compare_diff(reports1/html_path, reports2/html_path, verbose)
+
+def compare_json(proof, reports1, reports2,
+                 report='report', jsn='json', verbose=False):
+    # pylint: disable=too-many-arguments
+    """Compare json summaries produced by cbmc-viewer."""
+
+    json_path = proof/report/jsn
+    if not (reports1/json_path).is_dir():
+        raise FileNotFoundError(reports1/json_path)
+    if not (reports2/json_path).is_dir():
+        raise FileNotFoundError(reports2/json_path)
+
+    equal_json = True
+    logging.info("Comparing %s", json_path)
+
+    result_path = json_path/RESULT_JSON
+    logging.info("Comparing %s", result_path)
+    equal_json = compare_result(reports1/result_path, reports2/result_path) and equal_json
+
+    source_path = json_path/SOURCE_JSON
+    logging.info("Comparing %s", source_path)
+    equal_json = compare_source(reports1/source_path, reports2/source_path) and equal_json
+
+    for other in OTHER_JSON:
+        other_path = json_path/other
+        logging.info("Comparing %s", other_path)
+        equal_json = compare_diff(reports1/other_path, reports2/other_path, verbose) and equal_json
+
+    return equal_json
```

Comment:
In the future, we might want to use [difflib](https://docs.python.org/3/library/difflib.html) or [json-diff](https://pypi.org/project/json-diff/) instead of textual diffs. Shall we at least create a GitHub issue now so we can track this improvements in the future?